### PR TITLE
Rust: fixes for unit tests

### PR DESCRIPTION
**Pull request by david-allison-1**
Comments on the tests should explain the rationales. All changes non-functional. Splitting this off to keep the review load at a reasonable pace

#8988 

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java [hunk 1]
```diff
@@ -1784,6 +1786,9 @@ public SaveModel(Model model, ArrayList<Object[]> templateChanges) {
                     }
                 }
 
+                // required for Rust: the modified time can't go backwards, and we updated the model by adding fields
+                // This could be done better
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java:1790]: Oh wow. CardTemplateEditor just can't ever be easy can it...

**Review by mikehardy — DISMISSED** [review]
These all look fine to me - thanks!

**Comment by mikehardy** [comment]
merge at will I think

File: AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java (hunk 1 above)
Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java:1790]: This one scared me - the fix was much easier than I expected

**Review by mikehardy — APPROVED** [review]
still +1 on it / still merge at will IMHO :-)

**Comment by david-allison-1** [comment]
Down to 23 unit test failures with the V16 backend. Closer by the day, thanks so much for the reviews!

**Comment by mikehardy** [comment]
That's huge! Visualize the meme image of the entire theater standing up and clapping here since I'm mobile and can't paste it

**Pull request merged**